racker/salus-telemetry-model
# Support logical operations in task criteria

# What

Updates task expressions so that they can consist of logical or comparison expressions.

## How to test

See new event management pr - https://github.com/racker/salus-event-engine-management/pull/49

# Why

Allows us to migrate over all ele alarms, such as:

```
if (code != 200) {
  return CRITICAL;
}
if (body regex "error" != "") {
  return CRITICAL;
}
return OK;
```

which can now be combined to effectively be represented like

```
if (code != 200 || body regex "error" != "") return CRITICAL
```

This is needed since TickScript can only have one statement per task state.  This expands on ele's functionality by allowing nested logical expressions like

```
if ((code != 200 || code !=500) && body == "") {
  return CRITICAL;
}
```
